week3/binary/maverickchaser/Main.hs

```diff
@@ -0,0 +1,12 @@
+bin :: Int -> String
+bin 0 = ""
+bin x = bin (x `div` 2) ++ show (x `mod` 2)
+
+enumerate = zip [0..]
+
+dec :: String -> Integer
+dec = sum . map toAdd . enumerate . reverse
```

Comment:
if you wrote `toAdd power bit = 2^power * set bit`, you could've used `sum . zipWith toAdd [0..] . reverse`